kernelci/cli/user.py

```diff
@@ -55,57 +57,50 @@ def find(attributes, config, api, indent):
 @click.argument('username')
 @Args.config
 @Args.api
-@Args.indent
-def token(username, config, api, indent):
+@catch_http_error
+def token(username, config, api):
     """Create a new API token using a user name and password"""
-    password = getpass.getpass()
     configs = kernelci.config.load(config)
     api_config = configs['api'][api]
     api = kernelci.api.get_api(api_config)
-    user_token = api.create_token(username, password)
-    click.echo(json.dumps(user_token, indent=indent))
-
-
-@kci_user.group(name='password')
-def user_password():
-    """Manage user passwords"""
+    password = getpass.getpass()
+    api_token = api.create_token(username, password)
+    click.echo(api_token['access_token'])
 
 
 @kci_user.command(secrets=True)
-@click.option('--username')
-@click.option('--email')
-@click.option('--group', multiple=True, help="User group(s)")
+@click.argument('attributes', nargs=-1)
+@click.option('--username', help="Username of the user to update (admin only)")
 @Args.config
 @Args.api
-@Args.indent
-def update(username, email, config,  # pylint: disable=too-many-arguments
-           api, secrets, group, indent):
-    """Update own user account"""
-    user = {}
-    if username:
-        user['username'] = username
-    if email:
-        user['email'] = email
-    if group:
-        user['groups'] = group
-    if not user:
-        raise click.ClickException("Sorry, nothing to update")
+@catch_http_error
+def update(attributes, username, config, api, secrets):
+    """Update user account data with the provided attributes"""
+    fields = split_attributes(attributes)
+    if not fields:
+        click.echo("No user details to update.")
+        return
+
     configs = kernelci.config.load(config)
     api_config = configs['api'][api]
     api = kernelci.api.get_api(api_config, secrets.api.token)
-    data = api.update_user(user)
-    click.echo(json.dumps(data, indent=indent))
+    if username:
+        users = api.get_users({"username": username})
+        if not users:
+            raise click.ClickException(f"User not found: {username}")
+        user_id = users[0]['id']
+    else:
+        user_id = None
+    api.update_user(fields, user_id)
 
 
 @kci_user.command(secrets=True)
 @click.argument('username')
 @click.argument('email')
-@click.option('--group', multiple=True, help="User group(s)")
```

Comment:
Why `group` has been eliminated?

Reply:
Because it' just one of the many attributes you can update in a user profile.  With the rework of `split_attributes()` you can now pass `group=a group=b` and have groups `a` and `b` set for a user (pending further changes to separate the internal model with the database group id and the public schema with just group names on the API side, but that's beyond the scope of this PR).

Reply:
Yes, but there should be a provision for adding users to groups at the time of creating users.

Reply:
Why would this be required?  Is there a case where creating a user first, and then adding it to a group is not going to work?

Reply:
Of course, it is going to work. My point is if we can add groups in the same request while creating a user, why should we add an extra request for it?

Reply:
Because then, where do we stop if we want to be able to set up all the fields of a user when creating it?

We may soon have notification preferences, social network profile links, biography, profile picture... these are all at the same level as "groups", they are arbitrary properties.  I guess `man useradd` will show you what adding a Unix user account involves, but that's pretty much set in stone now and simpler than a web service user account.  Specifying the groups was required when `admin` had a special functional meaning, now groups don't really matter.  The only required fields now really are `email` and `username`.

Reply:
Basically, when we create a user account on any website, we usually can add all the available fields at the time of account creation. Currently, we just have one additional field i.e. groups. So I guess it is easy to include it in the `add` command as well.
Anyway, this is just another approach to solving the issue. We can go your way too. Merging this now to wrap this up.

Reply:
Yes, to be fair this is a bit of a grey area and somewhat subject to personal taste etc.  My approach here is to try and keep the number of commands and arguments to the bare minimum as I expect additional ones to be required when we hit issues in corner cases.  If you look at the legacy command line tools, they have lots of options and are barely usable by hand - only in scripts or Jenkins jobs.  If the new command line tools tend to become more complex too over the months then it helps to start with the simplest possible implementation that still provides all the required features.  No doubt this will need to be reworked a few times anyway as things evolve.